src/main/java/org/takes/facets/hamcrest/HmRsHeader.java

```diff
@@ -24,55 +24,70 @@
 package org.takes.facets.hamcrest;
 
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Iterator;
-import java.util.LinkedList;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Map.Entry;
+import java.util.List;
 import org.hamcrest.Description;
 import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
 import org.hamcrest.TypeSafeMatcher;
 import org.takes.Response;
-import org.takes.misc.EntryImpl;
 
 /**
  * Response Header Matcher.
  *
  * <p>This "matcher" tests given response header.
- * <p>The header name(s) should be provided in lowercase.
```

Comment:
In order to proceed with the task I decided that the code should be refactored somehow(just because it's too verbose and not really efficient). I found that refactoring task already exists(#480) already exists and relevant PR(#483) is available. Instead of modifying ineffective code(that most likely will be thrown away very soon) I used what I found in PR #483. May be it's not strictly correct but modifying actually outdated code is just wasting of time - finally it would lead to re-implementing my feature (it would be not a merge but rather re-implementation)